server/src/main/java/org/elasticsearch/common/lucene/store/InputStreamIndexInput.java

```diff
@@ -88,4 +88,26 @@ public synchronized void reset() throws IOException {
         indexInput.seek(markPointer);
         counter = markCounter;
     }
+
+    @Override
+    public long skip(long n) throws IOException {
+        long skipBytes = n;
```

Comment:
The `if (a + b > c) { b = c - a; }` pattern doesn't quite do the right thing in the presence of overflow. Probably not a big deal here given that we're dealing with `long` values, but I think we could avoid overflow concerns more thoroughly like this:

```suggestion
        long skipBytes = Math.min(n, Math.min(indexInput.length() - indexInput.getFilePointer(), limit - counter));
```

Reply:
Right, nice suggestion! It also makes it possible to make the checks much more succinct, since they all reduce to just checking skipBytes being negative. Thanks